lib/asset_cloud/metadata.rb

```diff
@@ -10,8 +10,13 @@ def exist?
       self.exist
     end
 
-    def initialize(exist, size = nil, created_at = nil, updated_at = nil, value_hash = nil)
-      self.exist, self.size, self.created_at, self.updated_at, self.value_hash = exist, size, created_at, updated_at, value_hash
+    def initialize(exist, size = nil, created_at = nil, updated_at = nil, value_hash = nil, checksum = nil)
```

Comment:
This currently causes a RuboCop issue, hence the change to `.rubocop_todo.yml`.

6 arguments is a bit on the high side. The [suggestion](https://learning.oreilly.com/library/view/refactoring-improving-the/9780134757681/ch03.xhtml#ch03lev1sec4) in Refactoring is to use  **Introduce Parameter Object** or **Preserve Whole Object**.

I think **Preserve Whole Object** isn't appropriate because it means passing a `StoredAsset`, which would couple the gem to one particular use case.

For **Introduce Parameter Object**, the fields don't really fall into any natural groupings, so the parameter object would probably contain all the fields, and we'd still end up with an initializer containing 6 fields.

Any thoughts @ayronshopify ?

Reply:
(as discussed on Slack, we'll leave this as-is)